### PR TITLE
[router] add pop/popToTop/canPop

### PR DESCRIPTION
**Pull request by marklawlor**
# Why

Add `pop`/`popToTop`/`canTop` to the router apis (both hooks and imperative).

# How

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/router/advanced/stack.mdx [hunk 1]
```diff
@@ -272,34 +272,69 @@ export default function Layout() {
...
+By default, the screens in a `Stack` are pushed or popped to the next or the previous screen. 
 
 For example, a stack has three screens &mdash; `home`, `details`, and `settings`. The following layout describes the order of the screens in a stack.
 
 <FileTree files={['app/_layout.js', 'app/home.js', 'app/details.js', 'app/settings.js']} />
 
+
+### `pop` action
+
+Go back in the stack by the number of screens specified in the `count` parameter. If no `count` is specified, then it will go back to the previous screen in the stack.
```
Comment (EvanBacon) [inline docs/pages/router/advanced/stack.mdx:286]: It should be clear how this differs from "goBack"

File: docs/pages/router/advanced/stack.mdx [hunk 2]
```diff
@@ -313,6 +348,43 @@ export default function Settings() {
...
+  return (
+    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
+      /* @info Trigger the handler method on a touchable/button component */
+      <Button title="Maybe pop" onPress={() => handlePop()} />
+      /* @end */
+    </View>
+  );
+}
+
+```
```
Comment (EvanBacon) [inline docs/pages/router/advanced/stack.mdx:386]: Maybe add a goBack vs pop example section in here.

**Comment by marklawlor** [comment]
Rename to `dismiss`

File: docs/pages/router/advanced/stack.mdx [hunk 3]
```diff
@@ -313,6 +348,43 @@ export default function Settings() {
 }
 ```
 
+### `canPop`
+
+To check if popping is available on the current route. Returns `true` is the router is within screen and there is a screen available to pop to.
```
Comment (amandeepmittal) [inline docs/pages/router/advanced/stack.mdx:353]:
```suggestion
To check if popping is available on the current route. Returns `true` if the router is within the screen and there is a screen available to pop to.
```

**Pull request merged**